netpyne/sim/utils.py

```diff
@@ -140,18 +140,17 @@ def timing(mode, processName):
         sim.timingData = {}
 
     if hasattr(sim.cfg, 'timing'):
-        if sim.cfg.timing:
```

Comment:
@salvadord/@joewgraham, from the PR description:
> All `if sim.cfg.timing: print()` are changed to `logger.timing()`.

This change made the code like `logger.timing('\nTotal time = %0.2f s' % sim.timingData['totalTime'])` fail when we have `sim.cfg.timing` set to `False`.

We can either always track time (this is what this commit does, thought we'd have to remove `hasattr(sim.cfg, 'timing')` too and do a couple of other things), or I can revert all timing statements to `if sim.cfg.timing: logger.info()`.

However - have we considered using a higher order function, say in this fashion https://stackoverflow.com/a/15136422/3192470?
Then we'd have it all in a single place:

```
import time

def timeit(f, message):
    def timed(*args, **kw):
        if sim.cfg.timing and sim.rank == 0:
          ts = time.time()
          result = f(*args, **kw)
          te = time.time()

          logger.info(message % (te-ts))
          return result
    return timed

@timeit(message="  Done; cells modification time =")
def modifyCells(self, params, updateMasterAllCells=False):
  ...
```
(Instead of https://github.com/Neurosim-lab/netpyne/blob/development/netpyne/network/modify.py)

Reply:
It's looking impressive, but I think we should discuss details at the dev meeting tomorrow.  I've added notes to the agenda.